src/rpc/methods/mpool.rs

```diff
@@ -146,6 +146,32 @@ impl RpcMethod<1> for MpoolPush {
     }
 }
 
+/// Add a batch of `SignedMessage`s to `mpool`, return message CIDs
+pub enum MpoolBatchPush {}
+impl RpcMethod<1> for MpoolBatchPush {
+    const NAME: &'static str = "Filecoin.MpoolBatchPush";
+    const PARAM_NAMES: [&'static str; 1] = ["msgs"];
+    const API_PATHS: ApiPaths = ApiPaths::V0;
+    /// Lotus limits this method to [`Permission::Write`].
+    /// However, since messages can always be pushed over the p2p protocol,
+    /// limiting the RPC doesn't improve security.
+    const PERMISSION: Permission = Permission::Read;
```

Comment:
This is only true if we have the same trust model for both types of access, which isn't true.

Consider this network topology for a node:
```console
$ ufw allow from 10.0.1.0/24 to any port 1234 proto tcp # allow RPC access from 10.0.1.x
$ ufw allow from 10.0.2.0/24 to any port # allow libP2P access from 10.0.2.x
```

If I trust `1.x` and `2.x` with different things, then this breaks my trust.